usr/lib/cca_stdll/cca_specific.c

```diff
@@ -285,6 +301,23 @@ static CK_RV init_cca_adapter_lock(STDLL_TokData_t *tokdata)
     pthread_rwlockattr_t attr;
     unsigned long cnt;
 
+    if (pthread_rwlockattr_init(&attr) != 0) {
```

Comment:
OK, seems that we forgot to initialize the min_card_version_rwlock lock ....
Anyway, I guess you don't need the pthread_rwlockattr_init() at all, you can do the `pthread_rwlock_init(&cca_private->min_card_version_rwlock, NULL);` with NULL instead of an attr.

Reply:
also this attr is for the cca_adapter_rwlock ock below, not for the min_card_version_rwlock lock. Looks like you have somehow mixed this up.